tests/e2e/multichain/utils.go

```diff
@@ -0,0 +1,276 @@
+
+	"go.uber.org/zap/zaptest"
+
+	"github.com/stretchr/testify/assert"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	ethenc "github.com/xpladev/ethermint/encoding/codec"
+)
+
+const (
+	wallet  = "xpla"
+	denom   = "axpla"
+	display = "XPLA"
+
+	minGasPrice = "280000000000"
+)
+
+var (
+	decimals = int64(18)
+
+	xplaDefaultBalance, _ = sdkmath.NewIntFromString("10_000_000_000_000_000_000")
+	simdDefaultBalance, _ = sdkmath.NewIntFromString("10_000_000")
+
+	LocalImage = []ibc.DockerImage{
+		{
+			Repository: "xpla",
+			Version:    "local",
+			UIDGID:     "1025:1025",
+		},
+	}
+)
+
+func XplaChainSpec(
+	numValidators int,
+	numFullNodes int,
+	chainID string,
+	version []ibc.DockerImage,
+) *interchaintest.ChainSpec {
+
+	genesis := []cosmos.GenesisKV{
+		cosmos.NewGenesisKV("app_state.gov.params.voting_period", "1m"),
+		cosmos.NewGenesisKV("app_state.gov.params.min_deposit.0.denom", denom),
+		cosmos.NewGenesisKV("app_state.gov.params.min_deposit.0.amount", "1"),
+		cosmos.NewGenesisKV("app_state.gov.params.expedited_min_deposit.0.denom", denom),
+		cosmos.NewGenesisKV("app_state.gov.params.expedited_min_deposit.0.amount", "1"),
+
+		cosmos.NewGenesisKV("app_state.mint.params.mint_denom", denom),
+		cosmos.NewGenesisKV("app_state.mint.minter.inflation", "0.000000000000000000"),
+		cosmos.NewGenesisKV("app_state.mint.params.inflation_rate_change", "0.000000000000000000"),
+		cosmos.NewGenesisKV("app_state.mint.params.inflation_min", "0.000000000000000000"),
+		cosmos.NewGenesisKV("app_state.mint.params.inflation_max", "0.000000000000000000"),
+
+		cosmos.NewGenesisKV("consensus.params.block.max_gas", "50000000000"),
+		cosmos.NewGenesisKV("app_state.evm.params.evm_denom", denom),
+		cosmos.NewGenesisKV("app_state.feemarket.params.min_gas_price", minGasPrice),
+
+		cosmos.NewGenesisKV("app_state.crisis.constant_fee.denom", denom),
+	}
+
+	encoding := wasm.WasmEncoding()
+	ethenc.RegisterInterfaces(encoding.InterfaceRegistry)
+
+	return &interchaintest.ChainSpec{
+		Name:          "xpla_1",
+		NumValidators: &numValidators,
+		NumFullNodes:  &numFullNodes,
+		ChainConfig: ibc.ChainConfig{
+			Name:             "xpla_1",
+			Type:             "cosmos",
+			ChainID:          chainID,
+			Images:           version,
+			Bin:              "xplad",
+			Bech32Prefix:     wallet,
+			Denom:            denom,
+			CoinType:         "60",
+			GasPrices:        minGasPrice + denom,
+			GasAdjustment:    1.5,
+			TrustingPeriod:   "168h0m0s",
+			ModifyGenesis:    cosmos.ModifyGenesis(genesis),
+			CoinDecimals:     &decimals,
+			SigningAlgorithm: "eth_secp256k1",
+			// open the port for the EVM on all nodes
+			ExposeAdditionalPorts: []string{"8545/tcp"},
+			EncodingConfig:        encoding,
+		},
+	}
+}
+
+func StartXplaChain(t *testing.T, ctx context.Context, version []ibc.DockerImage) (*cosmos.CosmosChain, *client.Client) {
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount("xpla", "xplapub")
```

Comment:
To set config once, it would be better move config getting and setting codes to init(). I think this could cause the race condition when concurrent testing.